### PR TITLE
Added UART2 UART3 PWM8-ch1ch2ch3ch4 and rt-robot package to the minibalance-diy

### PR DESCRIPTION
**Pull request by incarry**
## 拉取/合并请求描述：(PR description)

[

请在这里填写您的PR描述，可以包括以下之一的内容：为什么提交这份PR；解决的问题是什么，你的解决方案是什么；
Please fill in your PR description here, which can include one of the following items: why to submit this PR; what is the problem solved and what is your solution;

为什么：增加了UART2、UART3、PWM8-ch1ch2ch3ch4和rt-robot包。
Why: Added UART2 UART3 PWM8-ch1ch2ch3ch4 and rt-robot package.

并确认并列出已经在什么情况或板卡上进行了测试。
And confirm in which case or board have been tested.

测试：Shell测试
Tested: implement shell test.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/stm32/stm32f103-minibalance-diy/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+该开发板常用 **板载资源** 如下：
+
+- MCU：STM32F103RCT6，主频 72MHz，256KB FLASH ，48KB RAM
+- 常用外设
+  - LED：1个，L1（红色，PB13）
+  - 按键：2个，S1（用户自定义按键，PB14），S2（复位）
+- 常用接口：USB 转串口、OLED 接口、CAN 接口等
+- 调试接口: SWD
+
+开发板更多详细信息请参考【厂商名】 [minibalance开发板介绍](https://item.taobao.com/item.htm?spm=a1z09.2.0.0.5dcf2e8deKdZqx&id=550016227751&_u=p1sknvd56bd)。
```
Comment (BernardXiong) [inline bsp/stm32/stm32f103-minibalance-diy/README.md]: 请不要放广告

File: bsp/stm32/stm32f103-minibalance-diy/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+## 外设支持
+
+本 BSP 目前对外设的支持情况如下：
+
+| **板载外设**      | **支持情况** | **备注**                              |
+| :---------------- | :----------: | :-------------------------------------|
+| USB 转串口        |     支持     |                                       |
+| SPI Flash         |   暂不支持   |                                       |
+| 以太网            |   暂不支持   |                                       |
```
Comment (BernardXiong) [inline bsp/stm32/stm32f103-minibalance-diy/README.md]: 开发板中是否有对应的外设？如果没有，请保持表格简单、清晰

File: bsp/stm32/stm32f103-minibalance-diy/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+#### 运行结果
+
+下载程序成功之后，系统会自动运行，【这里写开发板运行起来之后的现象，如：LED 闪烁等】。
+
+连接开发板对应串口到 PC , 在终端工具里打开相应的串口（115200-8-1-N），复位设备后，可以看到 RT-Thread 的输出信息:
+
+```bash
+ \ | /
+- RT -     Thread Operating System
+ / | \     3.1.1 build Nov 19 2018
```
Comment (BernardXiong) [inline bsp/stm32/stm32f103-minibalance-diy/README.md]: 请给出正确的信息。

File: bsp/stm32/stm32f103-minibalance-diy/board/Kconfig [hunk 4]
```diff
@@ -0,0 +1,106 @@
+menu "Hardware Drivers Config"
+
+config SOC_STM32F103RC
+    bool
+	select SOC_SERIES_STM32F1
```
Comment (BernardXiong) [inline bsp/stm32/stm32f103-minibalance-diy/board/Kconfig]: 请注意对齐情况

**Comment by BernardXiong** [comment]
bsp/stm32/stm32f103-minibalance-diy/list_device

应该是一个空文件，请移除

**Comment by incarry** [comment]
> bsp/stm32/stm32f103-minibalance-diy/list_device
> 
> 应该是一个空文件，请移除

OK, Please

**Comment by BernardXiong** [comment]
这个修改提交太多了，还请合并成一份，然后重新提交PR，谢谢

**Pull request closed without merging**